tests/test_doctestplus.py

```diff
@@ -731,4 +732,38 @@ def test_remote_data(testdir):
     )
     testdir.inline_run(p, '--doctest-plus', '--doctest-rst').assertoutcome(skipped=1)
 
+    p = testdir.makefile(
+        '.rst',
+        """
+        
+        #This test is skipped when remote data is not requested, as it should
+        #but when remote data is requested the +FLOAT_CMP directive fails
+        .. doctest-remote-data::
+            >>> x = 1/3.
+            >>> x    # doctest: +FLOAT_CMP
+            0.333333
+            #
+            # do some failing to check whether the whole block is indeed skipped
+            >>> 1 + 1
+            3
+        """
+    )
+    testdir.inline_run(p, '--doctest-plus', '--doctest-rst', '--remote-data').assertoutcome(skipped=1)
+
+    p = testdir.makefile(
+        '.rst',
+        """
+        #This test should be skipped when remote data is not requested, but it doesn't. It gives 
+        #an assertion outcome of passed hence it fails
+        .. doctest-remote-data::
+            >>> import warnings
+            >>> warnings.warn('A warning occurred', UserWarning)  # doctest: +IGNORE_WARNINGS
+            
+        """
+    )
+    testdir.inline_run(p, '--doctest-plus', '--doctest-rst').assertoutcome(skipped=1)
```

Comment:
@saimn @pllim - we're kind of stuck here. This code chunck is skipped as expected with the new directive, but the test comes back as "passed" rather than "skipped". Have you recall any similar situation, or the reason for this?

Reply:
Are you sure it is really skipped? I installed this branch and manually ran one of the example above with `FLOAT_CMP` and it failed with or without `--remote-data` provided. I don't think the skipping logic is working.

Reply:
So, **after** I apply this patch myself, I see what you are saying.

```diff
--- a/pytest_doctestplus/plugin.py
+++ b/pytest_doctestplus/plugin.py
@@ -328,7 +328,7 @@ def pytest_configure(config):
                     if config.getoption('remote_data', 'none') != 'any':
                         print(config.getoption('remote_data', 'none') != 'any')
                         matches = [re.match(
-                            r'{}\s+doctest-remote-data\s*::(\s+.*)?'.format(comment_char),
+                            r'{}\s+doctest-remote-data\s*::'.format(comment_char),
                             last_line) for last_line in last_lines]
 
                         if len(matches) > 1:
@@ -337,12 +337,8 @@ def pytest_configure(config):
                             match = matches[0]
 
                         if match:
-                            marker = match.group(1)
-                            if (marker is None or
-                                    (marker.strip() == 'win32' and
-                                    sys.platform == 'win32')):
-                                skip_next = True
-                                continue
+                            skip_next = True
+                            continue
```

I think `# doctest: +IGNORE_WARNINGS` is messing with the remote data logic. If you remove it, it is skipped as expected. Not sure what the best fix for that is.

Reply:
Hmm, it might have been in a yet uncommitted version we discussed during the call, there it was certainly skipped, but came back as "passed".

Reply:
I don't understand, on the branch it is skipped.

Reply:
I explained in https://github.com/astropy/pytest-doctestplus/issues/147#issuecomment-760983505

Reply:
I think the problem is now resolved.